### PR TITLE
support upload generated file and report to blob

### PR DESCRIPTION
**Pull request by ruowan**

### CODE REVIEW

File: lib/testScenario/_testPostmanReportParser.ts [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 const main = async () => {
   try {
-    const parser = new PostmanReportParser(
-      "//home/ruowan/work/oav/newman/newman-run-report-2021-02-09-07-44-08-992-0.json",
-      "./generated_reports/generated_report.json"
-    );
-    parser.generateRawReport();
+    const opts: NewmanReportAnalyzerOption = {
+      newmanReportFilePath:
+        "/home/ruowan/work/oav/newman/newman-run-report-2021-02-09-07-44-08-992-0.json",
```
Comment (raych1) [inline lib/testScenario/_testPostmanReportParser.ts:11]: Can you remove this debugging code?

File: lib/testScenario/blobUploader.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  private blobServiceClient?: BlobServiceClient;
+  // eslint-disable-next-line @typescript-eslint/explicit-member-accessibility
+  constructor(@inject(TYPES.opts) private opts: TestScenarioBlobUploaderOption) {
+    setDefaultOpts(this.opts, {
+      enableBlobUploader: false,
+      blobConnectionString: "",
+    });
+    if (opts.enableBlobUploader) {
+      this.blobServiceClient = BlobServiceClient.fromConnectionString(
+        this.opts.blobConnectionString!
```
Comment (raych1) [inline lib/testScenario/blobUploader.ts:23]: How do we get the connectionString? we can use environment variable firstly then consider to use keyvault.

File: lib/commands/generate-collection.ts [hunk 3]
```diff
@@ -80,6 +85,8 @@ export async function handler(argv: yargs.Arguments): Promise<void> {
         outputFolder: argv.output,
         eraseXmsExamples: false,
         eraseDescription: false,
+        enableBlobUploader: false,
```
[the remark this answers is not in the document]
Comment (ruowan) [reply lib/commands/generate-collection.ts:88]: collection file is for debugging purpose. We need to save collection files to blob.

[the remark this answers is not in the document]
Comment (raych1) [reply lib/commands/generate-collection.ts:88]: Does it contain credentials saved in env?

[the remark this answers is not in the document]
Comment (ruowan) [reply lib/commands/generate-collection.ts:88]: will mask credential data.

File: lib/testScenario/_testPostmanReportParser.ts (hunk 1 above)
Comment (ruowan) [reply lib/testScenario/_testPostmanReportParser.ts:11]: remove later

File: lib/testScenario/blobUploader.ts (hunk 2 above)
Comment (ruowan) [reply lib/testScenario/blobUploader.ts:23]: connect to keyvalut will be a seprate step.

**Review by raych1 — APPROVED** [review]

**Pull request merged**